app/services/location_maps_generator_service.rb

```diff
@@ -20,22 +20,39 @@ def generate
     end
 
     lib_display = {}
-    longer_display = {}
+    locations_display = {}
     holding_library = {}
     locations.each do |holding|
       holding_code = holding['code']
       lib_label = holding.library['label']
-      holding_label = holding['label'] == '' ? lib_label : lib_label + ' - ' + holding['label']
+      holding_label = holding['label'] == '' ? lib_label : holding['label']
       lib_display[holding_code] = lib_label
-      longer_display[holding_code] = holding_label
-      holding_library[holding_code] = holding.holding_library['label'] if holding.holding_library
+      locations_display[holding_code] = holding_label
+    end
+
+    # The Alma::Locations misses the location code: rare$warf
+    # As a result the locations.rb file does not include it.
+    # Do we need to add this location code ?
+    # It refers to: Western Americana: Reference Collection (WARF)
+    holding_libraries.each do |library|
+      holding_library[library.code] = library.label
     end
 
     write_file('location_display.rb', lib_display)
-    write_file('locations.rb', longer_display)
+    write_file('locations.rb', locations_display)
     write_file('holding_library.rb', holding_library)
   end
 
+  # Select the holding_libraries based on the holding location code from:
+  # https://github.com/pulibrary/bibdata/blob/main/marc_to_solr/translation_maps/holding_library.rb.tmpl
```

Comment:
Are these the locations that aren't in alma? Can you just add to the comment that they're not in alma because they are maintained by ReCAP's inventory management system? (I think that's why)

Reply:
These are locations in Alma. The template file which is currently in production includes only these locations codes. 
I think it is related to a comment we added a while ago in traject about the holding library. `The holding_library is used with some locations to add an additional owning library, which is included in advanced search but not facets.` This is the file that traject is reading from (holding_library.tmpl). https://github.com/pulibrary/bibdata/blob/6a0a6b6d5d4f991e4e959b0a774933e7ea688b42/marc_to_solr/lib/traject_config.rb#L1193

Reply:
I see. Okay I guess my question then is why do we prefer to list them out this way? Were they not pulling out of alma with the check for `holding.holding_library`?

Reply:
No they were not. The way the code was set up in the alma branch would generate an empty holding_library.rb file.
Also holding.holding_library was nil in the alma branch. My guess is that the holding_library.tmpl file was a copy and paste from the production file with an update on the location_codes based on the alma criteria. At least this is what I came across when I tried to generate the files.

Reply:
@mzelesky said that this file holding_library.tmpl shows which library handles items for these locations. Currently this is not maintained in Alma. @mzelesky is investigating if this can be implemented in Alma. But for now I will update the code to include the hash as it is in production with the new location codes.